packages/plugins/i18n/server/services/entity-service-decorator.js

```diff
@@ -167,18 +166,14 @@ const decorator = (service) => ({
 
     const { kind } = strapi.getModel(uid);
 
-    const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });
-
-    const query = transformParamsToQuery(uid, wrappedParams);
-
     if (kind === 'singleType') {
       if (opts[LOCALE_QUERY_FILTER] === 'all') {
-        return strapi.db.query(uid).findMany(query);
+        return service.findMany.call(this, uid, { ...opts, ignoreKind: true });
```

Comment:
I don't like the idea.

but since it unblocks it sure.

What makes ingoreKind so hard to review?

Reply:
Agree, we still will have the bug.

We need to be sure of the steps we take on sanitization and entity service, as those changes will be there to stay. And also ben has been very busy this past week :)

Reply:
for sanitization this change should not matter since it is done before it and after it 

And findMany or findOne in this cease would have the same sanitization done on the controller level so it should not matter. Still I understand that this would have to be double and triple checked since we don't want to introduce a CVE.

Reply:
The issue is only that because it's adding a feature, once it's there it can't be changed without being a breaking change, so I want to make sure it's the best way. If you don't mind waiting until Friday, you can leave the PR as-is and I can get an internal decision made then :)

Reply:
hello ! 
We continued discussions about this work, and we want to keep exploring other solutions for ignoreKind, specially to make it conceptually work with V5. As we discussed with you, we might need to change the behaviour of findMany for singleTypes in V5, so we are not really sure we want the `ignoreKind` flag.

So we think we could split that part away from this PR , and continue the discussion about `ignoreKind` in another PR. This wont fully solve the bug but it's a start and we move things forward.

wdyt @Boegie19 ?

Reply:
I don't see an other none breaking way out
But I will split off the pr into 2 prs like you asked.
So it is atleast a lot less broken